tools/docker-network/tests/coreapi_test.go

```diff
@@ -0,0 +1,542 @@
+			testFunc: func(t *testing.T, nodeAlias string) {
+				assetsPerSlot.forEachOutput(t, func(t *testing.T, outputID iotago.OutputID, output iotago.Output) {
+					if output.Type() != iotago.OutputDelegation {
+						return
+					}
+
+					resp, err := d.wallet.Clients[nodeAlias].Rewards(context.Background(), outputID)
+					require.NoError(t, err)
+					require.NotNil(t, resp)
+					// rewards are zero, because we do not wait for the epoch end
+					require.EqualValues(t, 0, resp.Rewards)
+				})
+			},
+		},
+		{
+			name: "Test_Committee",
+			testFunc: func(t *testing.T, nodeAlias string) {
+				resp, err := d.wallet.Clients[nodeAlias].Committee(context.Background())
+				require.NoError(t, err)
+				require.NotNil(t, resp)
+				require.EqualValues(t, 4, len(resp.Committee))
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			d.requestFromClients(test.testFunc)
+		})
+	}
+
+	// check if the same values were returned by all nodes for the same slot
+	assetsPerSlot.assertCommitments(t)
+	assetsPerSlot.assertBICs(t)
+}
+
+func Test_CoreAPI_BadRequests(t *testing.T) {
+	d := NewDockerTestFramework(t,
+		WithProtocolParametersOptions(
+			iotago.WithTimeProviderOptions(5, time.Now().Unix(), 10, 4),
+			iotago.WithLivenessOptions(10, 10, 2, 4, 8),
+			iotago.WithRewardsOptions(8, 10, 2, 384),
+			iotago.WithTargetCommitteeSize(4),
+		))
+	defer d.Stop()
+
+	d.AddValidatorNode("V1", "docker-network-inx-validator-1-1", "http://localhost:8050", "rms1pzg8cqhfxqhq7pt37y8cs4v5u4kcc48lquy2k73ehsdhf5ukhya3y5rx2w6")
+	d.AddValidatorNode("V2", "docker-network-inx-validator-2-1", "http://localhost:8060", "rms1pqm4xk8e9ny5w5rxjkvtp249tfhlwvcshyr3pc0665jvp7g3hc875k538hl")
+	d.AddValidatorNode("V3", "docker-network-inx-validator-3-1", "http://localhost:8070", "rms1pp4wuuz0y42caz48vv876qfpmffswsvg40zz8v79sy8cp0jfxm4kunflcgt")
+	d.AddValidatorNode("V4", "docker-network-inx-validator-4-1", "http://localhost:8040", "rms1pr8cxs3dzu9xh4cduff4dd4cxdthpjkpwmz2244f75m0urslrsvtsshrrjw")
+	d.AddNode("node5", "docker-network-node-5-1", "http://localhost:8090")
+
+	runErr := d.Run()
+	require.NoError(t, runErr)
+
+	d.WaitUntilNetworkReady()
+
+	tests := []struct {
+		name     string
+		testFunc func(t *testing.T, nodeAlias string)
+	}{
+		{
+			name: "Test_BlockByBlockID_Failure",
+			testFunc: func(t *testing.T, nodeAlias string) {
+				blockID := tpkg.RandBlockID()
+				respBlock, err := d.wallet.Clients[nodeAlias].BlockByBlockID(context.Background(), blockID)
+				require.Error(t, err)
+				require.True(t, isStatusCode(err, http.StatusNotFound))
+				require.Nil(t, respBlock)
+			},
+		},
+		{
+			name: "Test_BlockMetadataByBlockID_Failure",
+			testFunc: func(t *testing.T, nodeAlias string) {
+				blockID := tpkg.RandBlockID()
+				resp, err := d.wallet.Clients[nodeAlias].BlockMetadataByBlockID(context.Background(), blockID)
+				require.Error(t, err)
+				require.True(t, isStatusCode(err, http.StatusInternalServerError))
```

Comment:
This should return NotFound?

Reply:
It was like that, but this still can return error for getting the store, so it's better to use InternalServerErr, see the above MuXxer comment